### ISSUE TITLE
Can’t reproduce the high cosine similarity result in the paper with shared ACFNet checkpoint

### ISSUE THREAD

**Issue by sg47** [jiaw-z/FeatureConsistentStereo#1]
Hi, Thank you for sharing! 
Follow the GETTING_STARED.md,  i implemented the cosine similarity myself and checked it on the demo data with the checkpoint from https://github.com/DeepMotionAIResearch/DenseMatchingBenchmark/blob/master/configs/AcfNet/ResultOfAcfNet.md#sceneflow
and most of the feature similarity are still around 0.6-0.9.
Is the checkpoint provided already trained for feature consistency? 
And it would be appreciated if a script to check the similarity could be provided.
Thanks again!

**Comment by jiaw-z** [comment]
Hi, sorry to reply late.
The checkpoint is trained by authors of DenseMatchingBenchmark, so it is trained without our feature consistency constraints.
We are about to release our pre-trained checkpoints later.
You can train models with consistency constraints with the config file in [./configs/FCStereo/](https://github.com/jiaw-z/FCStereo/blob/main/configs/FCStereo).
The similarity reported in the paper is calculated on valid and non-occ pixels with pixel-wise cosine similarity. The valid masks are generated in the same way as the valid masks for contrastive loss, 
![image](https://user-images.githubusercontent.com/66359549/162685950-ae12e506-b4d3-4140-afd5-cea7878ecdda.png)
which can be found in[./dmb/modeling/stereo/losses/contrastive_loss.py](https://github.com/jiaw-z/FCStereo/blob/main/dmb/modeling/stereo/losses/contrastive_loss.py).
Hope this can help you!

**Comment by mattpoggi** [comment]
> We are about to release our pre-trained checkpoints later.

Hi, any news about this?

**Comment by Zhaohuai-L** [comment]
Hi, can you share your pretrained checkpoint?

**Comment by jiaw-z** [comment]
> Hi, can you share your pretrained checkpoint?

We are very sorry that the sharing of checkpoints may be delayed. We used a lot of storage when experimenting with our work for cvpr2023 and accidentally deleted all our previous checkpoints. We need to retrain models to obtain checkpoints for FCStereo.

**Comment by colorfulgreen** [comment]
[NOTICE] Zhang Jiawei(Gary) deleted my workspace without notice, which containes data and network models of a paper under review.

<img width="558" alt="image" src="https://github.com/jiaw-z/FCStereo/assets/2982465/71caf29e-01b0-499c-abf3-d26f000faedc">
